File: day2/day2.py
games = [ [ord(_[0])-65, ord(_[-1])-88] for _ in open("input.txt").read().splitlines()]

part1 = sum([ 3 * ((b - a + 1) % 3) + b + 1 for a, b in games ])
part2 = sum([ (a + b + 2) % 3 + 1 + b * 3 for a, b in games ])

print("Part 1:", part1)
print("Part 2:", part2)

File: day2/day2.rs
use std::fs;

fn main() {
    let file:String = fs::read_to_string("input.txt").unwrap();
    let lines:Vec<&str> = file.split("\n").collect();

    let games:Vec<[i32; 2]> = lines.iter().map( |line:&&str| {
        [line.chars().nth(0).unwrap() as i32 - 65, line.chars().nth(2).unwrap() as i32 - 88]
    }).collect();

    let part1:i32 = games.iter().map( | [a,b]:&[i32;2] | (3 * ((b - a + 4) % 3)) + b + 1 ).sum();
    let part2:i32 = games.iter().map( | [a,b]:&[i32;2] | (a + b + 2) % 3 + 1 + b * 3 ).sum();

    println!("Part 1: {}", part1);
    println!("Part 2: {}", part2);
}